src/components/Picker/Picker.tsx

```diff
@@ -105,6 +106,12 @@ class Picker extends React.Component<Props, State> {
       number: 0,
     };
   }
+
+  componentWillReceiveProps(newProps: Props) {
+    if (newProps.source.length !== this.props.source.length) {
+      this.setState({ itemsList: newProps.source });
+    }
```

Comment:
Here logic looks wrong, what will happen if source is different object array but of same length as prev props source.
We should check if nextProps.source array is different of props.source array.